### PR TITLE
Port Colony Motions

### PR DESCRIPTION
**Pull request by willm30**
## Description

Feature branch for Motions.

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

### CODE REVIEW

**Review by ArmandoGraterol — APPROVED** [review]
This is the most remarkable feature branch I've ever seen if I do say so myself

**Review by Julianahlk — COMMENTED** [review]
Overall, the usual tests seem to be working although I noticed 2 edge cases that may or may not be something new

1. I have a motion that gets stuck in the Reveal state, reveal ends in 0s and the reveal button remains disabled. I am using 270 for the time in the extension and waiting for the time to pass instead of fast forwarding time.
Screenshot 2023-06-06 at 1 04 46 PM
Screenshot 2023-06-06 at 1 05 19 PM

2. I have a motion that was fully staked and objected by User B then with User A I was able to vote and reveal even though they did not stake the motion. Now when I log in with User B, its showing reveal and when i click reveal, I get an error that the User did not submit their vote and there is no way now to vote. Eventually the motion does move to finalize and can be completed.
Screenshot 2023-06-06 at 1 08 25 PM

**Comment by willm30** [comment]
@Julianahlk 

> I have a motion that gets stuck in the Reveal state,

Let's test the timer on QA as it's possible your block time isn't moving (e.g. if you have the rep monitor turned off).

> I have a motion that was fully staked and objected by User B then with User A

Being able to vote on a motion you didn't stake is expected. I imagine what's happening is you're switching between users quickly and before the client has finished sending messages to the chain / block ingestor. I expect you can't reproduce this if you take a more leisurely pace 😉

**Review by rdig — APPROVED** [review]

**Review by jakubcolony — APPROVED** [review]
Let's get it merged! 💪

**Review by Julianahlk — APPROVED** [review]
> Let's test the timer on QA as it's possible your block time isn't moving (e.g. if you have the rep monitor turned off).

> Being able to vote on a motion you didn't stake is expected. I imagine what's happening is you're switching between users quickly and before the client has finished sending messages to the chain / block ingestor. I expect you can't reproduce this if you take a more leisurely pace 😉

Sounds good 👌

File: src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/useStakingSlider.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import { useAppContext } from '~hooks';
+import { getTransactionHashFromPathName } from '~utils/urls';
+
+import { calculateStakeLimitDecimal } from '../helpers';
+import { useEnoughTokensForStaking } from './useEnoughTokensForStaking';
+
+export const useStakingSlider = (isObjection: boolean) => {
+  const { user, userLoading, walletConnecting } = useAppContext();
+  const { pathname } = useLocation();
+  const transactionHash = getTransactionHashFromPathName(pathname);
```
[the remark this answers is not in the document]
Comment (willm30) [reply src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/useStakingSlider.ts:14]: Yes, can't recall why useParams doesn't work, but did notice it wasn't returning the tx hash as expected
